### PR TITLE
Solr changes to get random sound

### PR DESCRIPTION
**Pull request by andrebola**
This PR is linked with the issues #631 and #593

Regarding changes on Solr to allow range queries on the id field for the API, also added some other attributes to solr to get a list of random sounds.

### CODE REVIEW

File: utils/search/search_general.py [hunk 1]
```diff
@@ -39,6 +41,7 @@ def convert_to_solr_document(sound):
     document["description"] = remove_control_chars(sound.description)
     document["tag"] = list(sound.tags.select_related("tag").values_list('tag__name', flat=True))
     document["license"] = sound.license.name
+    document["is_explicit"] = sound.is_explicit
```
Comment (alastair) [inline utils/search/search_general.py]: does this have to be added to the schema?

Comment (andrebola) [inline utils/search/search_general.py]: yes!

File: sounds/views.py [hunk 2]
```diff
@@ -125,10 +127,18 @@ def remixed(request):
...
 
 def random(request):
-    sound_id = Sound.objects.random()
-    if sound_id is None:
+    sounds = get_random_sounds_from_solr(1)
+    sound_obj = None
+    if sounds:
+        try:
+            sound_obj = Sound.objects.get(id=sounds[0]['id'])
+        except ObjectDoesNotExist:
```
Comment (alastair) [inline sounds/views.py]: `Sound.DoesNotExist`

File: sounds/views.py [hunk 3]
```diff
@@ -125,10 +127,18 @@ def remixed(request):
 
 
 def random(request):
-    sound_id = Sound.objects.random()
-    if sound_id is None:
+    sounds = get_random_sounds_from_solr(1)
```
Comment (alastair) [inline sounds/views.py]: Will this method be used in any other places? I wonder if we don't need to get multiple random sounds - what if this method only returns 1 sound?

File: sounds/views.py [hunk 4]
```diff
@@ -125,10 +127,18 @@ def remixed(request):
...
+    sound_obj = None
+    if sounds:
+        try:
+            sound_obj = Sound.objects.get(id=sounds[0]['id'])
+        except ObjectDoesNotExist:
+            pass
+    if sound_obj is None:
+        sound_id = Sound.objects.random()
+        sound_obj = Sound.objects.get(pk=sound_id)
+    if sound_obj is None:
```
Comment (alastair) [inline sounds/views.py]: The line above is going to raise `Sound.DoesNotExist` if it doesn't work, not return `None`

File: sounds/views.py [hunk 5]
```diff
@@ -125,10 +127,18 @@ def remixed(request):
...
-    if sound_id is None:
+    sounds = get_random_sounds_from_solr(1)
+    sound_obj = None
+    if sounds:
+        try:
+            sound_obj = Sound.objects.get(id=sounds[0]['id'])
+        except ObjectDoesNotExist:
+            pass
+    if sound_obj is None:
+        sound_id = Sound.objects.random()
```
Comment (alastair) [inline sounds/views.py]: This method has been updated to return an object, not an id: https://github.com/MTG/freesound/blob/b66b51d7c5294eca737fcc44f3c55aad1e786c33/sounds/models.py#L103

It'd be nice to have a test for this view which we know would catch this change.

**Comment by andrebola** [comment]
After this changes the sound accessed after going to http://freesound.org/browse/random should be retrieved using solr and not the DB. There should not be any change on the behavior of the random sound shown in the frontpage

**Comment by alastair** [comment]
This looks better :+1:
After talking with @ffont we decided to release this schema on appserver3, then we can reindex and switch over without downtime, then I will upgrade search to debian jessie and we can move the index back. I'll take care of the release.

**Pull request merged**